ldk/node/examples/consolidatedExample/src/index.ts

```diff
@@ -0,0 +1,293 @@
+    };
+    const decoder = new TextDecoder();
+    const encoder = new TextEncoder();
+
+    try {
+      this.host.fileSystem.createFile('./data.txt');
+      this.host.keyboard.streamHotKey(hotkeys, (error, keyboardResponse) => {
+        if (error) {
+          logger.error(error);
+        } else {
+          this.host.whisper.formWhisper(formConfig, (e, formResponse) => {
+            if (e !== null) {
+              // Error occured
+            }
+
+            if (typeof formResponse === 'undefined') {
+              logger.info('Form response error');
+            }
+
+            const updateEvent = formResponse as WhisperFormUpdateEvent;
+            const submitEvent = formResponse as WhisperFormSubmitEvent;
+
+            if (updateEvent.type === 'update') {
+              // do nothing for now
+              // logger.info('update Occurred');
+              // logger.info(JSON.stringify(updateEvent));
+            } else if (submitEvent.type === 'submit' && submitEvent.submitted) {
+              const formErrors = this.validateForm(submitEvent);
+              if (formErrors.length === 0) {
+                const readHandle = this.host.fileSystem.openFile('./data.txt');
+
+                readHandle
+                  .read()
+                  .then((fileContents) => {
+                    const file = decoder.decode(fileContents);
+                    logger.info(`${file === ''}`);
+                    const fileObj = file === '' ? {} : JSON.parse(file);
+                    fileObj[
+                      submitEvent.outputs.email.toString()
+                    ] = JSON.stringify(submitEvent.outputs);
+                    logger.info(JSON.stringify(fileObj));
+
+                    const writeHandle = this.host.fileSystem.openFile(
+                      './data.txt',
+                    );
+                    return writeHandle.write(
+                      encoder.encode(JSON.stringify(fileObj)),
+                    );
+                  })
+                  .then(() => {
+                    logger.info('success');
+                  })
+                  .catch((err) => {
+                    logger.error(err);
+                  });
+              } else {
+                logger.info('fail');
+                this.host.whisper.markdownWhisper({
+                  label: 'There was an issue with the form data',
+                  markdown: formErrors,
+                });
+              }
+            } else {
+              logger.info('cancelled');
+            }
+          });
+        }
+      });
+    } catch (e) {
+      logger.error('Error With Keyboard Stream', 'error', e.toString());
+    }
+  }
+
+  startGlobalSearchSensor(): void {
+    const decoder = new TextDecoder();
+
+    try {
+      this.host.ui.streamSearchbar((error, searchTerm) => {
+        if (error) {
+          logger.error('Eek!');
+        } else if (searchTerm) {
+          // TODO: Right now, requires pressing enter twice, see SIDE-1597
```

Comment:
SIDE-1597 is merged. Is anything else left to do? Do we need a card to take care of anything extra?